### PR TITLE
Window manager flags and events for SDL

### PR DESCRIPTION
**Pull request by Efimero**

### CODE REVIEW

File: core/SDL.carp [hunk 1]
```diff
@@ -56,6 +59,12 @@
     (register init (Fn [] SDL_Event))
     (register type (Fn [(Ref SDL_Event)] SDL_EventType))
     (register keycode (Fn [(Ref SDL_Event)] SDL_Keycode))
+    ;(register window (Fn [(Ref SDL_Event)] SDL_WindowEvent))
```
Comment (hellerve) [inline core/SDL.carp:62]: Should the commented version stay in there?

File: core/SDL.carp [hunk 2]
```diff
@@ -205,6 +215,66 @@
...
+      (register window-hit-test     SDL_WindowEventID "SDL_WINDOWEVENT_HIT_TEST")
+
+      (register = (Fn [SDL_WindowEventID SDL_WindowEventID] Bool))
+      (defn = [a b]
+        (Int.= (enum-to-int (the SDL_WindowEventID a))
+               (enum-to-int (the SDL_WindowEventID b))))
+      (implements = SDL.WindowEvent.WindowEventID.=)
+      (register copy (Fn [(Ref SDL_WindowEventID)] SDL_WindowEventID))
+      (implements copy SDL.WindowEvent.WindowEventID.copy))
+    ;(register event (Fn [SDL_WindowEvent] SDL_WindowEventID))
```
Comment (hellerve) [inline core/SDL.carp:245]: Same as above!

File: core/SDL.carp [hunk 3]
```diff
@@ -205,6 +215,66 @@
...
+      (register window-restored     SDL_WindowEventID "SDL_WINDOWEVENT_RESTORED")
+      (register window-enter        SDL_WindowEventID "SDL_WINDOWEVENT_ENTER")
+      (register window-leave        SDL_WindowEventID "SDL_WINDOWEVENT_LEAVE")
+      (register window-focus-gained SDL_WindowEventID "SDL_WINDOWEVENT_FOCUS_GAINED")
+      (register window-focus-lost   SDL_WindowEventID "SDL_WINDOWEVENT_FOCUS_LOST")
+      (register window-close        SDL_WindowEventID "SDL_WINDOWEVENT_CLOSE")
+      (register window-take-focus   SDL_WindowEventID "SDL_WINDOWEVENT_TAKE_FOCUS")
+      (register window-hit-test     SDL_WindowEventID "SDL_WINDOWEVENT_HIT_TEST")
+
+      (register = (Fn [SDL_WindowEventID SDL_WindowEventID] Bool))
```
Comment (hellerve) [inline core/SDL.carp:238]: You don’t need to register it if you define it yourself, only if it’s defined elsewhere (i.e. in C)

**Review by hellerve — APPROVED** [review]
This looks good, thanks for your work!

I have a few minor comments, but they’re not deal breakers at all!

File: core/SDL.carp (hunk 1 above)
Comment (Efimero) [reply core/SDL.carp:62]: Yes, because the intent is to not rely on deftemplate in the future. I'll add a note.

Comment (Efimero) [inline core/SDL.carp:62]:
```suggestion
    ;This is intentionally left here to note that in the future we shouldn't need deftemplate for this.
    ;(register window (Fn [(Ref SDL_Event)] SDL_WindowEvent))
```

File: core/SDL.carp (hunk 2 above)
Comment (Efimero) [inline core/SDL.carp:245]:
```suggestion
    ;This is intentionally left here to note that in the future we shouldn't need deftemplate for this.
    ;(register event (Fn [SDL_WindowEvent] SDL_WindowEventID))
```

File: core/SDL.carp (hunk 3 above)
Comment (Efimero) [inline core/SDL.carp:238]:
```suggestion
```

**Review by hellerve — APPROVED** [review]
Looks great, thanks for the clarifying comments!

**Comment by eriksvedang** [comment]
Sweet!

**Pull request merged**